packages/flutter/lib/src/cupertino/text_form_field_row.dart

```diff
@@ -0,0 +1,391 @@
+    String? placeholder,
+    TextStyle? placeholderStyle = const TextStyle(
+      fontWeight: FontWeight.w400,
+      color: CupertinoColors.placeholderText,
+    ),
+  })  : assert(initialValue == null || controller == null),
+        assert(textAlign != null),
+        assert(autofocus != null),
+        assert(readOnly != null),
+        assert(obscuringCharacter != null && obscuringCharacter.length == 1),
+        assert(obscureText != null),
+        assert(autocorrect != null),
+        assert(enableSuggestions != null),
+        assert(maxLengthEnforced != null),
+        assert(scrollPadding != null),
+        assert(maxLines == null || maxLines > 0),
+        assert(minLines == null || minLines > 0),
+        assert(
+          (maxLines == null) || (minLines == null) || (maxLines >= minLines),
+          "minLines can't be greater than maxLines",
+        ),
+        assert(expands != null),
+        assert(
+          !expands || (maxLines == null && minLines == null),
+          'minLines and maxLines must be null when expands is true.',
+        ),
+        assert(!obscureText || maxLines == 1,
+            'Obscured fields cannot be multiline.'),
+        assert(maxLength == null || maxLength > 0),
+        assert(enableInteractiveSelection != null),
+        super(
+          key: key,
+          initialValue:
+              controller != null ? controller.text : (initialValue ?? ''),
+          onSaved: onSaved,
+          validator: validator,
+          autovalidateMode: autovalidateMode,
+          builder: (FormFieldState<String> field) {
+            final _CupertinoTextFormFieldRowState state =
+                field as _CupertinoTextFormFieldRowState;
+
+            void onChangedHandler(String value) {
+              field.didChange(value);
+              if (onChanged != null) {
+                onChanged(value);
+              }
+            }
+
+            return CupertinoFormRow(
+              prefix: prefix,
+              margins: margins,
+              error: (field.errorText == null) ? null : Text(field.errorText!),
+              child: CupertinoTextField.borderless(
+                controller: state._effectiveController,
+                focusNode: focusNode,
+                keyboardType: keyboardType,
+                decoration: decoration,
+                textInputAction: textInputAction,
+                style: style,
+                strutStyle: strutStyle,
+                textAlign: textAlign,
+                textAlignVertical: textAlignVertical,
+                textCapitalization: textCapitalization,
+                autofocus: autofocus,
+                toolbarOptions: toolbarOptions,
+                readOnly: readOnly,
+                showCursor: showCursor,
+                obscuringCharacter: obscuringCharacter,
+                obscureText: obscureText,
+                autocorrect: autocorrect,
+                smartDashesType: smartDashesType ??
```

Comment:
nit: I think we can just pass in `smartDashesType` as-is here, since the `CupertinoTextField` constructor knows how to deal with a null value. Same with `smartQuotesType`.